lib/icinga/downtime.cpp

```diff
@@ -328,11 +328,15 @@ Downtime::Ptr Downtime::AddDowntime(const Checkable::Ptr& checkable, const Strin
 
 	if (!triggeredBy.IsEmpty()) {
 		Downtime::Ptr parentDowntime = Downtime::GetByName(triggeredBy);
-		Array::Ptr triggers = parentDowntime->GetTriggers();
 
-		ObjectLock olock(triggers);
-		if (!triggers->Contains(fullName))
-			triggers->Add(fullName);
+		if (parentDowntime) {
```

Comment:
Yes, this fixes the null dereference, but wouldn't it make more sense to throw an exception if a non-existing downtime is referenced (and therefore move it to before actually creating the downtime object)? Otherwise, otherwise, it would be an option that provides no feedback if you use it wrong.

Reply:
Then the API would reject something I can already assemble via config or even the API itself:

1. Schedule Downtime A
2. Schedule Downtime B with trigger_name=A
3. Remove Downtime A
4. Reload Icinga (optional)

Downtime B still exists with trigger_name=A. But the world doesn't burn or similar.

Reply:
It might be possible, but is there a good reason why anyone would want to do this?

Reply:
As I said, (4) is optional. And (3) ~could~ _will_ happen by itself, just due to expiry. I.e. if you simply use the trigger_name feature as designed, you _will_ end up with an invalid Downtime#triggered_by on Downtime B once Downtime A expires. I mean... I get your point. But, I'm afraid, Downtime#triggered_by="no such Downtime" is perfectly valid.

Reply:
> And (3) ~could~ _will_ happen by itself, just due to expiry. I.e. if you simply use the trigger_name feature as designed, you _will_ end up with an invalid Downtime#triggered_by on Downtime B once Downtime A expires.

There's a difference between having a dangling reference as a leftover and initializing objects with a dangling reference. Which purpose does is serve once it became dangling? If there's none, why would you want to create it this way?